torch/csrc/jit/passes/create_autodiff_subgraphs.cpp

```diff
@@ -177,17 +264,18 @@ class SubgraphSlicer {
     return isDifferentiable(node);
   }
 
-  std::pair<graph_node_list::iterator, bool> scanNode(
-      Node* consumer,
-      AliasDb& aliasDb) {
+  std::pair<graph_node_list::iterator, bool> scanNode(Node* consumer) {
     if (shouldConsiderForMerge(consumer)) {
       if (consumer->kind() != prim::DifferentiableGraph) {
+        // ValueMapper preserves the aliasing information of the node's outputs
+        ValueMapper vm(consumer, aliasDb_, 0);
```

Comment:
nitpick: we could probably just remember the number of the outputs of a producer (which we can get off the producer rather than pass both producer and the offset) since we will need to update aliasing for all of them and then in a consumer we could just update the aliasing information for the last  `consumer.outputs().size() - producer.outputs().size()` of the merged consumer?

Reply:
That's what I'm doing, see: https://github.com/pytorch/pytorch/pull/42141/files#diff-d35d4ac08adc5a54ab5b75564e9a1290R53